### ISSUE TITLE
Metadata for training

### ISSUE THREAD

**Issue by nicoaira** [nicoaira/StrusiNet2#9]
Con cada training generar un json con metadatos sobre el training:

id del modelo (cada configuracion de modelos deberia tener tambien un ID unico)
tiempo de entrenamiento
hiperparametros (epochs, learning rate, decay rate, etc)
plosts de resultados (val loss y train loss vs epochs) (opcional)
plataforma (sistema operativo, gpu, etc). Aca un ejemplo de como recoger estos datos (se deberian ajustar para que terminen en un json):

```
import platform
import psutil
import os
import GPUtil

def get_system_info():
    # Operating System
    os_name = platform.system()
    os_version = platform.version()
    os_release = platform.release()

    # CPU Info
    cpu = platform.processor()
    physical_cores = psutil.cpu_count(logical=False)
    total_cores = psutil.cpu_count(logical=True)

    # Memory Info
    svmem = psutil.virtual_memory()
    total_memory = svmem.total / (1024 ** 3)  # Convert to GB

    # Disk Info
    total_disk_space = psutil.disk_usage('/').total / (1024 ** 3)  # Convert to GB

    # GPU Info
    gpus = GPUtil.getGPUs()
    gpu_info = []
    for gpu in gpus:
        gpu_info.append({
            "Name": gpu.name,
            "Load": f"{gpu.load * 100:.2f}%",
            "Memory Total": f"{gpu.memoryTotal}MB",
            "Memory Used": f"{gpu.memoryUsed}MB",
            "Memory Free": f"{gpu.memoryFree}MB",
            "Driver Version": gpu.driver
        })

    # Collect all info into a dictionary
    system_info = {
        "Operating System": f"{os_name} {os_release} (Version: {os_version})",
        "CPU": {
            "Name": cpu,
            "Physical Cores": physical_cores,
            "Total Cores": total_cores,
        },
        "Memory": f"{total_memory:.2f} GB",
        "Disk Space": f"{total_disk_space:.2f} GB",
        "GPU": gpu_info if gpu_info else "No GPU Detected",
    }

    return system_info

def print_system_info(system_info):
    print("System Information:")
    print("===================")
    print(f"Operating System: {system_info['Operating System']}")
    print(f"CPU: {system_info['CPU']['Name']}")
    print(f"  - Physical Cores: {system_info['CPU']['Physical Cores']}")
    print(f"  - Total Cores: {system_info['CPU']['Total Cores']}")
    print(f"Memory: {system_info['Memory']}")
    print(f"Disk Space: {system_info['Disk Space']}")

    if system_info['GPU'] != "No GPU Detected":
        print("GPU Information:")
        for idx, gpu in enumerate(system_info['GPU']):
            print(f"  GPU {idx + 1}: {gpu['Name']}")
            print(f"    - Load: {gpu['Load']}")
            print(f"    - Memory Total: {gpu['Memory Total']}")
            print(f"    - Driver Version: {gpu['Driver Version']}")
    else:
        print("GPU: No GPU Detected")

system_info = get_system_info()
print_system_info(system_info)
```

**Comment by mekicastro** [comment]
Agregué un log para tanto el training como el prediction que imprime toda la información respecto a la corrida, tanto datos de plataforma como de ejecución e hiper-parametros.
El formato lo hice copiando el formato que usaste para loggear en benchmark.py, que transforma todo en un json y después lo va imprimiendo durante la ejecución.
La idea es en vez de esto tener todo en un .json en vez de un .log? O queremos tener ambos? @nicoaira
